src/test/lib/utils/list_directory_test.cpp

```diff
@@ -0,0 +1,81 @@
+#include <filesystem>
+#include <fstream>
+#include <unordered_set>
+#include <random>
+#include <limits>
+
+#include "base_test.hpp"
+
+#include "utils/list_directory.hpp"
+
+namespace hyrise {
+
+class ListDirectoryTest : public BaseTest {
+ protected:
+  ListDirectoryTest() {
+    // You can do set-up work for each test here.
+    auto rand_dev = std::random_device();
+    auto generator = std::mt19937();
+    auto distribution = std::uniform_int_distribution<uint32_t>(0, INT_MAX);
+
+    _directory_path.append(std::to_string(distribution(generator)));
+  }
+
+  void SetUp() override {
+    std::filesystem::create_directory(_directory_path);
+  }
+
+  void TearDown() override {
+    std::filesystem::remove_all(_directory_path);
+  }
+
+  std::string _directory_path = test_data_path + "/test-directory";
+};
+
+TEST_F(ListDirectoryTest, ListDirectoryWithFilesAndSubdirectory) {
+  const auto files = std::unordered_set<std::string>{_directory_path + "/one.test", _directory_path + "/two.test",
+                                                     _directory_path + "/three.test", _directory_path + "/four.test"};
```

Comment:
```suggestion
  const auto files = std::vector<std::string>{_directory_path + "/one.test", _directory_path + "/two.test",
                                              _directory_path + "/three.test", _directory_path + "/four.test"};
```

Reply:
This was changed to an unordered set following a suggestion [here](https://github.com/hyrise/hyrise/pull/2576#discussion_r1231117267) by Daniel. I also think it is a more elegant implementation, since we later can use `contains(..)` to look up the values. 

I was first a little worried, since `contains(..)` is introduced in C++20, but it seems the pipeline likes it anyway.